src/krishna/Shop.java

```diff
@@ -0,0 +1,89 @@
+package krishna;
+
+public class Shop {
+	String maggie="Maggie",dosa= "Dosa",pouches = "Pouches",panipuri = "Panipuri",masala= "Masala";
+	int maggieQuan=50,dosaQuan=43,pouchesQuan=39,panipuriQuan=43,masalaQuan=73;
+	
+	void setInitial(int maggieQ,int dosaQ,int pouchesQ,int panipuriQ,int masalaQ) {
+		if(maggieQ<=50) {
+			maggieQuan = maggieQuan-maggieQ;
+			System.out.println("Available stock "+maggie+": " +maggieQuan);
+		}else{
+			maggieQuan=0;
+			System.out.println("Maggie running out of stock");
```

Comment:
Why we printing the OOS statements here, we have separate requirement for this FR2 and FR3